ex/high_throughput_inference/mock_app.py

```diff
@@ -101,43 +108,49 @@ def run_model(self, model: bytes | str, batch: torch.Tensor):
         self.perf_timer.measure_time("build_request")
         request_bytes = MessageHandler.serialize_request(request)
         self.perf_timer.measure_time("serialize_request")
-        with self._to_worker_fli.sendh(timeout=None, stream_channel=self._to_worker_ch) as to_sendh:
+        with self._to_worker_fli.sendh(
+            timeout=None, stream_channel=self._to_worker_ch
+        ) as to_sendh:
             to_sendh.send_bytes(request_bytes)
             self.perf_timer.measure_time("send_request")
             for tensor in tensors:
-                to_sendh.send_bytes(tensor.tobytes()) #TODO NOT FAST ENOUGH!!!
+                to_sendh.send_bytes(tensor.tobytes())  # TODO NOT FAST ENOUGH!!!
         self.perf_timer.measure_time("send_tensors")
-        with self._from_worker_ch.recvh(timeout=None) as from_recvh:
-            resp = from_recvh.recv_bytes(timeout=None)
-            self.perf_timer.measure_time("receive_response")
-            response = MessageHandler.deserialize_response(resp)
-            self.perf_timer.measure_time("deserialize_response")
-            # list of data blobs? recv depending on the len(response.result.descriptors)?
-            data_blob: bytes = from_recvh.recv_bytes(timeout=None)
-            self.perf_timer.measure_time("receive_tensor")
-            result = torch.from_numpy(
-                numpy.frombuffer(
-                    data_blob,
-                    dtype=str(response.result.descriptors[0].dataType),
-                )
+
+        resp = self._from_worker_ch.recv(timeout=None)
+        self.perf_timer.measure_time("receive_response")
+        response = MessageHandler.deserialize_response(resp[0])
+        self.perf_timer.measure_time("deserialize_response")
+        # list of data blobs? recv depending on the len(response.result.descriptors)?
+        if len(resp) > 1:
+            data_blob = resp[1]
+        else:
+            data_blob: bytes = self._from_worker_ch.recv(timeout=None)[0]
+        self.perf_timer.measure_time("receive_tensor")
+        result = torch.from_numpy(
```

Comment:
I think a change I had made a while ago got reversed in one of the merges! I had changed it so that `result` is a `numpy array` instead of a `torch tensor`. Should we go back to that or is a torch tensor what we need to return now?

Reply:
Totally right, I must have missed that! Let me revert it to numpy array!